gui/src/WorkspaceView/controls/SurfaceSelectionControls.tsx

```diff
@@ -0,0 +1,164 @@
+        }
+    })), [surfacesWithFields, selection.selectedSurfaceVectorFieldNames, selection.selectedSurfaceScalarFieldNames, selection.visibleSurfaceNames])
+
+    return (
+        <div key="surfaces">
+            <h3>Surfaces</h3>
+            {
+                surfaces.map(surface => (
+                    <PerSurfaceControls
+                        name={surface.name}
+                        displayed={surface.displayed}
+                        scalarFields={surface.scalarFields}
+                        vectorFields={surface.vectorFields}
+                        selectedScalarFieldName={surface.selectedScalarFieldName}
+                        selectedVectorFieldName={surface.selectedVectorFieldName}
+                        visibilityCallback={handleToggleVisibleSurface}
+                        selectedScalarFieldCallback={handleToggleSelectedScalarField}
+                        selectedVectorFieldCallback={handleToggleSelectedVectorField}
+                    />
+                ))
+            }
+        </div>
+    )
+}
+
+const PerSurfaceControls: FunctionComponent<PerSurfaceControlsProps> = (surface: PerSurfaceControlsProps) => {
+    const { name, displayed, scalarFields, vectorFields, selectedScalarFieldName, selectedVectorFieldName, visibilityCallback, selectedVectorFieldCallback, selectedScalarFieldCallback } = surface
+    return (
+        <div key={name}>
+            <h3 key="surface">Surface: {name}</h3>
+            <div key="show">
+                <Checkbox
+                    value={name}
+                    checked={displayed}
+                    onChange={visibilityCallback}
+                /> show surface
+            </div>
+            {/* Vector field selector */}
+            {displayed && <FieldSelector
+                surfaceName={name}
+                fields={vectorFields}
+                selectedFieldName={selectedVectorFieldName}
+                type={'vector'}
+                callback={selectedVectorFieldCallback}
+            />}
+            {/* Scalar field selector */}
+            {displayed && <FieldSelector
+                surfaceName={name}
+                fields={scalarFields}
+                selectedFieldName={selectedScalarFieldName}
+                type={'scalar'}
+                callback={selectedScalarFieldCallback}
+            />}
+        </div>
+    )
+}
+
+const FieldSelector: FunctionComponent<FieldSelectorProps> = (props: FieldSelectorProps) => {
+    const { surfaceName, fields, selectedFieldName, type, callback } = props
+    if (fields.length === 0) return (<span />)
+    return (
+        <div key={`surface-${type}-fields`}>
+            <span>
+                <h4>Surface {type} fields</h4>
+                {
+                    fields.map(field => (
+                        <div key={field.name}>
+                            <Checkbox
+                                value={`${surfaceName}${separator}${field.name}`}
```

Comment:
Here's where the checkbox value encodes the surface name along with the field name.

Everything is parameterized by the global separator value, so it's straightforward to change the separator to a different constant or to generate some random string that isn't part of any of the allocated names in the data set.